src/hwlesson16/task5/Main.java

```diff
@@ -0,0 +1,53 @@
+package hwlesson16.task5;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Main {
+    public static void main(String[] args) {
+
+        /**
+         * Задать список учеников с принадлежностью к классу (создать класс Student).
+         * Используя средства Stream API
+         * 1) вывести список фамилий
+         * 2) создать Map со списком учеников для каждого класса
+         * */
+
+        //creating students
+        Student student1 = new Student("Anna", 1);
+        Student student2 = new Student("Barbara", 1);
+        Student student3 = new Student("Victoria", 1);
+        Student student4 = new Student("Glen", 2);
+        Student student5 = new Student("Maks", 2);
+        Student student6 = new Student("Tim", 2);
+
+        //creating group list and add students
+        List<Student> students = new ArrayList<>();
+        students.add(student1);
+        students.add(student2);
+        students.add(student3);
+        students.add(student4);
+        students.add(student5);
+        students.add(student6);
+
+        //print students names
+        students.stream().forEach((s)-> System.out.println(s.getName()));
+
+        //creating list for group1
+        List<Student> group1 = students.stream().filter((student) -> student.getGroup() == 1).collect(Collectors.toList());
```

Comment:
it is can be like one way, but usual when we write some decision, we try use universal solution, like we don't know about value of field. like additional material you can read https://metanit.com/java/tutorial/10.7.php